tests/manage/z_cluster/nodes/test_node_replacement_proactive.py

```diff
@@ -0,0 +1,60 @@
+import logging
+import re
+
+import pytest
+import random
+from ocs_ci.framework.testlib import (
+    tier4, ManageTest, aws_platform_required, ignore_leftovers, ipi_deployment_required)
+from ocs_ci.ocs import machine
+from ocs_ci.ocs import constants
+from ocs_ci.ocs import ocp
+from ocs_ci.ocs import node
+from tests.sanity_helpers import Sanity
+
+log = logging.getLogger(__name__)
+
+
+@tier4
+@ignore_leftovers
+@aws_platform_required
+@ipi_deployment_required
+class Testknip894(ManageTest):
+    """
+    Knip-678 Node Replacement
+    """
+
+    @pytest.fixture(autouse=True)
+    def init_sanity(self):
+        """
+        Initialize Sanity instance
+        """
+        self.sanity_helpers = Sanity()
+
+    def test_nodereplacement_proactive(self, pvc_factory, pod_factory):
+        """
+        Knip-894 Node Replacement proactive
+        """
+        self.sanity_helpers.create_resources(pvc_factory, pod_factory)
+        ocs_worker_node = node.get_ocs_worker_nodes()
+        machine_name = machine.get_machine_from_node_name(random.choice(ocs_worker_node))
```

Comment:
This logic may fail and the test case may not be valid incase of 4 worker nodes and all labeled case. 
Instead you can get the random node of a osd.
You can refer to test_automated_recovery_from_failed_nodes_proactive_IPI.py
  osd_pods_obj = pod.get_osd_pods()
        osd_node_obj = pod.get_pod_node(random.choice(osd_pods_obj))
        log.info(f"Selected OSD is {osd_node_obj.name}")

        # Get the machine name using the node name
        machine_name = machine.get_machine_from_node_name(osd_node_obj.name)
        log.info(f"{osd_node_obj.name} associated machine is {machine_name}")
By which we will make sure that we are deleting a node where osd is present.

Reply:
will update in next commit